treeherder/services/taskcluster.py

```diff
@@ -0,0 +1,137 @@
+
+        # Following least-privilege principle, as services
+        # bellow don't really need authorization credentials.
+        self.queue = taskcluster.Queue(options)
+        self.auth = taskcluster.Auth(options)
+
+    def set_root_url(self, root_url):
+        for service in (self.hooks, self.queue, self.auth):
+            service.options['rootUrl'] = root_url
+
+    def trigger_action(self, action, task_id, decision_task_id, input, root_url=None) -> str:
+        if root_url is not None:
+            self.set_root_url(root_url)
+
+        actions_context = self._load(decision_task_id, task_id)
+        action_to_trigger = self._get_action(actions_context['actions'], action)
+
+        return self._submit(
+            action=action_to_trigger,
+            decision_task_id=decision_task_id,
+            task_id=task_id,
+            input=input,
+            static_action_variables=actions_context['staticActionVariables']
+        )
+
+    def _load(self, decision_task_id: str, task_id: str) -> dict:
+        if not decision_task_id:
+            raise ValueError("No decision task, can't find taskcluster actions")
+
+        # fetch
+        logger.debug('Fetching actions.json...')
+        actions_json = self.queue.getLatestArtifact(decision_task_id, 'public/actions.json')
+        task_definition = self.queue.task(task_id)
+
+        if actions_json['version'] != 1:
+            raise RuntimeError('Wrong version of actions.json, unable to continue')
+
+        return {
+            'staticActionVariables': actions_json['variables'],
+            'actions': self._filter_relevant_actions(actions_json, task_definition),
+        }
+
+    def _submit(self,
+                action=None,
+                decision_task_id=None,
+                task_id=None,
+                input=None,
+                static_action_variables=None) -> str:
+        context = {
+                "taskGroupId": decision_task_id,
+                "taskId": task_id or None,
+                "input": input,
+         }
+        context.update(static_action_variables)
+
+        if action["kind"] == "task":
+            raise NotImplementedError("Unable to submit actions with 'task' kind.")
+
+        if action["kind"] == "hook":
+            hook_payload = jsone.render(action["hookPayload"], context)
+            hook_id, hook_group_id = action["hookId"], action["hookGroupId"]
+
+            decision_task = self.queue.task(decision_task_id)
+            expansion = self.auth.expandScopes({"scopes": decision_task["scopes"]})
+            expression = f"in-tree:hook-action:{hook_group_id}/{hook_id}"
+
+            if not satisfiesExpression(expansion["scopes"], expression):
+                raise RuntimeError(f"Action is misconfigured: decision task's scopes do not satisfy {expression}")
+
+            result = self.hooks.triggerHook(hook_group_id, hook_id, hook_payload)
+            return result["status"]["taskId"]
```

Comment:
what if action['kind'] not in ['task', 'hook'], we don't return a value and this is supposed to return a string;  

Maybe check for hook and return, otherwise raise notimplementederror?